### ISSUE TITLE
Likelihood Of Test Set

### ISSUE THREAD

**Issue by surbut** [surbut/matrix_ash#12]
Here, I divide the data into a test and training set, each containing 8000 genes.

The covariance matrices are 'learned' from the strongest snp for the first 8000 genes using my deconvolution.em.with.loop framework.

The full set of 1188 covariance matrices are then produced after adding the output of step 1 to a mixture containing 5 single ranks and 44+1 'configs' and expanded to included an autoselected 22 grid weights.

The model is then 'trained' on a random set of 20,000 gene snp pairs restricted to these first 8000 genes.

The model likelihood is computed on a 'test set' consisting of a random set of 20,000 gene snp pairs restricted to the final 8000 genes.

![loglikelihoodtestset](https://cloud.githubusercontent.com/assets/4492893/9144436/8fdf57e4-3d12-11e5-878c-2030e51ce384.png)

**Comment by surbut** [comment]
As you can see, rank 3 produces the best results.

I then recompute the denoised covariance matrices using all 16,069 'strong' gene snp pairs, produce the list of covariance matrices, train the model on a randomly selected set of gene snp pairs from the whole genome and compute posteriors. This is available on midway under

/home/surbut/jul3/aug3rank3/analysis.html

**Comment by surbut** [comment]
Updated Plot. You can see that using single ranks that have been denoised does NOT improve the likelihood, so best to stick with denoising multiple rank matrices and then adding to the set of single rank, bam-lite and Identity matrices: 
![likelihoodcompare](https://cloud.githubusercontent.com/assets/4492893/9552821/8269b342-4d80-11e5-892d-3ca47af678b6.png)
